server/src/main/java/com/cloud/api/ApiSessionListener.java

```diff
@@ -55,29 +52,12 @@ public void sessionCreated(HttpSessionEvent event) {
         }
         LOGGER.debug("Sessions count: " + getSessionCount());
     }
+
     public void sessionDestroyed(HttpSessionEvent event) {
         LOGGER.debug("Session destroyed by Id : " + event.getSession().getId() + " , session: " + event.getSession().toString() + " , source: " + event.getSource().toString() + " , event: " + event.toString());
         synchronized (this) {
             sessions.remove(event.getSession());
         }
         LOGGER.debug("Sessions count: " + getSessionCount());
     }
-
-    @Override
-    public void requestDestroyed(ServletRequestEvent event) {
-        LOGGER.debug("request destroyed");
-    }
-
-    @Override
-    public void requestInitialized(ServletRequestEvent event) {
-        LOGGER.debug("request initialized");
-        HttpServletRequest request = (HttpServletRequest) event.getServletRequest();
-        HttpSession session = request.getSession();
-        if (session.isNew()) {
```

Comment:
@DaanHoogland maybe one way is to use a ConcurrentHashMap to store total unique sessions?

Reply:
not sure what you mean. It is already used like that , is it?